src/inc/til/winrt.h

```diff
@@ -0,0 +1,192 @@
+// Copyright (c) Microsoft Corporation.
+// Licensed under the MIT license.
+
+#pragma once
+
+namespace til // Terminal Implementation Library. Also: "Today I Learned"
+{
+    namespace details
+    {
+        template<typename T>
+        constexpr T winrt_empty_value() noexcept
+        {
+            if constexpr (std::is_base_of_v<winrt::Windows::Foundation::IUnknown, T>)
+            {
+                return nullptr;
+            }
+            else
+            {
+                return {};
+            }
+        }
+    };
+
+    template<typename T>
+    struct property
+    {
+        property<T>(const T& defaultValue = details::winrt_empty_value<T>()) :
+            _value{ defaultValue } {}
+
+        property<T>& operator=(const property<T>& other) = default;
+
+        T operator()() const
+        {
+            return _value;
+        }
+        void operator()(const T& value)
+        {
+            _value = value;
+        }
+        property<T>& operator=(const T& newValue)
+        {
+            _value = newValue;
+            return *this;
+        }
+        operator bool() const requires std::is_same_v<T, winrt::hstring>
+        {
+            return !_value.empty();
+        }
+        operator bool() const requires std::convertible_to<T, bool>
+        {
+            return _value;
+        }

```

Comment:
It's much cheaper for the compiler if you use `if constexpr` over any alternatives. I assume it's because the contents of functions (like templates) are only evaluated when used (unlike concepts). Like so:
```suggestion
        operator bool() const noexcept
        {
            if constexpr (std::is_same_v<T, winrt::hstring>)
            {
                return !_value.empty();
            }
            else
            {
                return _value;
            }
        }
```